lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -16,56 +18,71 @@ def make_logger(server_config):
 
 
 class TestUnpackTarballs:
-    tar_name = "A.tar.xz"
-    tarlist = [tar_name]
-    linksrc = "TO-UNPACK"
-
-    def mock_glob(self, path):
-        """Returns the Tarball list"""
-        return TestUnpackTarballs.tarlist
-
-    def mock_resolve(self: Path, strict: bool = False):
-        """
-        Return a "full" fake file name
-
-        Args:
-            strict: Not used in the mock
-
-        Returns:
-            new Path with fake resolution
-        """
-        return Path(self.tar_name)
+    tarlist = ["/mock/ABC/TO-UNPACK/A.tar.xz", "/mock/ABC/TO-UNPACK/B.tar.xz"]
 
     def test_tarball_symlinks(self, monkeypatch, server_config, make_logger, caplog):
         """Tests for an incorrect symlink which results in an unresolved location"""
 
-        monkeypatch.setattr(glob, "iglob", self.mock_glob)
-        obj = Unpack(server_config, self.linksrc, make_logger)
-        total, success = obj.unpack_tarball()
-        assert (
-            f"Tarball link, '{self.tar_name}', does not resolve to a real location"
-            in caplog.text
-        )
+        def mock_resolve(self: Path, strict: bool = False):
+            """
+            Return a "full" fake file name
+
+            Args:
+                strict: Not used in the mock
+
+            Returns:
+                new Path with fake resolution
+            """
+            raise FileNotFoundError
```

Comment:
I'm a huge fan of docstrings, but, sadly, _this_ docstring describes what the `Path.resolve()` function does, rather than what this **mock** for it does.

So, the function description should probably be something along the lines of, "Mock replacement for Path.resolve()."  And the "Returns" section should probably be replaced by a comment saying that the function always reports file-not-found.

The "Args" section is more interesting.  `Path.resolve()` doesn't raise `FileNotFoundError` if `strict` is `False` or unspecified -- that check is only done when explicitly requested.  So, the mock should probably be asserting that `strict` is true before doing the `raise`.  (Note that we don't need to do this in _all_ the mocks for `Path.resolve()` -- since all the tests operate on the same CUT, doing it in one of them should be sufficient...unless there is some reason to do it in the others...which I haven't spotted, yet.)

And, as Dave said, while Python will let us get away with `raise`'ing an `Exception` (sub)class, we really should raise a class instance, so that the code which catches the exception actually has some content (i.e., a file name or message) to play with (like you did at line 64).